lib/src/line_writer.dart

```diff
@@ -180,10 +180,11 @@ class LineWriter {
   void _writeChunksUnsplit(BlockChunk block) {
     for (var chunk in block.children) {
       if (chunk.spaceWhenUnsplit) _buffer.write(' ');
-      _writeChunk(chunk);
 
       // Recurse into the block.
       if (chunk is BlockChunk) _writeChunksUnsplit(chunk);
+
+      _writeChunk(chunk);
```

Comment:
This was an unrelated bug. But I think it never surfaced because you couldn't get into a state where you had an inner block that split inside a block that did not.